fairscale/optim/oss.py

```diff
@@ -382,16 +387,95 @@ def _broadcast_state_dict(self) -> None:
             else:
                 global_rank = self.get_global_rank(self.group, rank)
                 # Discard this tensor/rank, broadcast necessary for syncing
-                logging.debug("Discarding broadcast from rank %s", global_rank)
                 broadcast_object(empty_buffer, src_rank=global_rank, group=self.group, dist_device=self._device)
 
     def _free_other_grads(self) -> None:
         """Free all the gradients only useful for the other ranks
         """
-        for i, partition in enumerate(self.partition_parameters()):
-            if i == self.rank:
+        for rank, partition in enumerate(self.partition_parameters()):
+            if rank == self.rank:
                 continue
 
             for p in partition:
                 for t in p["params"]:
                     t.grad = None
+
+    @staticmethod
+    def get_global_rank(group: Any, rank: int) -> int:
+        if group is dist.group.WORLD:
+            return rank
+        else:
+            global_rank = dist.distributed_c10d._get_global_rank(group, rank)  # type: ignore
+        return global_rank
+
+    @staticmethod
+    def _consume_async_work(queue: List[Any]) -> None:
```

Comment:
Nit: might be more readable to just inline this.

Reply:
yeah, in the beginning the spirit was "let's remove all this duplicated code", now that everything is unrolled I agree that I can at least make it consistent. Taste and colors, I personally prefer tight functions in general so that the structure is more apparent